model/fn-execution/src/main/proto/beam_fn_api.proto

```diff
@@ -215,17 +215,21 @@ message Metrics {
   message PTransform {
     // Metrics that are measured for processed and active element groups.
     message Measured {
-      // (Required) Map from local input name to number of elements processed
+      // (Optional) Map from local input name to number of elements processed
       // from this input.
+      // If unset, assumed to be the sum of the outputs of all producers to
+      // this transform.
+      // Should be set for operations (such as reading from the
```

Comment:
Would you expect that a source would set this as the number of elements read from an external system?
What about a sink for output to an external system?

Also, what purpose do you see by recording the input count for `urn:org.apache.beam:source:runner:0.1` and the output count for `urn:org.apache.beam:sink:runner:0.1`?

Reply:
No, these counters are not settable by UserFns. 

The input counter for `urn:org.apache.beam:source:runner:0.1` may be useful to understand how much of the stream has been consumed. (I suppose it's equal to its output counter, modulo some lag.) The output counter for `urn:org.apache.beam:sink:runner:0.1` could be the number of elements written to the stream (again, this would be equal to its input counter == upstream output counters, with some lag). But maybe it's more consistent to leave them unset. Should I go ahead and do this?